crates/ruff_linter/src/checkers/tokens.rs

```diff
@@ -118,11 +118,14 @@ pub(crate) fn check_tokens(
         );
     }
 
+    if settings.rules.enabled(Rule::AvoidableEscapedQuote) && settings.flake8_quotes.avoid_escape {
+        flake8_quotes::rules::avoidable_escaped_quote(&mut diagnostics, tokens, locator, settings);
+    }
```

Comment:
What's the reason for moving the rule out of `from_tokens`?

Reply:
The idea was that the escaped quote check needs to be done in the content value only i.e., for f-string it's only the `FStringMiddle` tokens. 

The f-string context which needs to be tracked for this rule is different than the one for the quote rules where the main difference is how implicitly concatenated strings are handled. For this rule, implicit string concatenation doesn't matter because we only care about the content while for the quote rules it matters.

For quote rules, each sequence needs to be independently built. For nested f-strings, the nested sequence will be different than the outer one and so on. This is the main reason for the separation. It becomes easier to reason about.